### PR TITLE
ARROW-13544 [Java]: Remove APIs that have been deprecated for long (Changes to Vectors)

### PR DESCRIPTION
**Pull request by liyafan82**
See https://issues.apache.org/jira/browse/ARROW-13544

According to the discussion in #10864 (comment), we want to split the task into multiple parts.

This PR is for the changes related to vectors

### CODE REVIEW

File: java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java [hunk 1]
```diff
@@ -680,7 +659,7 @@ public void reset() {
 
   @Override
   public UnionVector promoteToUnion() {
-    UnionVector vector = new UnionVector("$data$", allocator, callBack);
+    UnionVector vector = new UnionVector("$data$", allocator, null, callBack);
```
Comment (emkornfield) [inline java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java:662]: please comment null literal.

**Comment by emkornfield** [comment]
Generally looks good, it looks like a lot of null literals where added in, it would be nice to comment those.  Thanks for the cleanup!

File: java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java (hunk 1 above)
Comment (liyafan82) [reply java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java]: Done. Thanks.

**Comment by liyafan82** [comment]
> Generally looks good, it looks like a lot of null literals where added in, it would be nice to comment those. Thanks for the cleanup!

Comments added. Please check. Thanks.

**Comment by liyafan82** [comment]
@emkornfield Thanks for reviewing this PR. Do you have any more comments?

**Comment by emkornfield** [comment]
Nope looks OK to me.

**Comment by emkornfield** [comment]
+1

**Pull request closed without merging**